src/compiler/parser.ts

```diff
@@ -2867,9 +2869,13 @@ namespace Parser {
                 return tokenIsIdentifierOrKeyword(token()) || token() === SyntaxKind.OpenBraceToken;
             case ParsingContext.JsxChildren:
                 return true;
+            case ParsingContext.JSDocComment:
+                return true;
+            case ParsingContext.Count:
+                return Debug.fail("ParsingContext.Count used as a context"); // Not a real context, only a marker.
+            default:
+                Debug.assertNever(parsingContext, "Non-exhaustive case in 'isListElement'.");

```

Comment:
how did this (mostly) work before? Do we never call`isListElement` inside jsdoc? (Unlikely). Did `currentNode(parsingContext)` always return a value, so the switch never executed? Was there some other incorrect parsing context set during jsdoc parsing, except for the bug case?

Reply:
This code was actually never hit for JSDoc, because the loop in `isInSomeParsingContext` checks each bit and _then_ uses it. Unlike source files (which are always inside of a statement list), the standalone JSDoc parser was _not_, and so its parsing context was zero, and skipped all of this code (which was the bug).